keps/sig-node/2033-rootless/README.md

```diff
@@ -0,0 +1,831 @@
+
+#### Story 3: `kind` with Rootless Docker/Podman
+
+A user wants to run a test cluster inside Docker/Podman on his/her laptop using `kind`.
+
+However, the user doesn't want Kubernetes/kind/Docker/Podman to gain the root privileges
+because these components may accidentally break the host environment,
+e.g. Docker may modify the host iptables in an unexpected way and break the user's VPN connectivity.
+
+The user would want this KEP to be implemented so that he/she could run `kind` with Rootless Docker/Podman,
+which won't break the host.
+
+#### Story 4: Temporary initial cluster for bootstrapping
+
+A user needs a temporary initial cluster to bootstrap an actual cluster with Cluster API.
+
+The user wants to avoid having the root privileges.
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+TL;DR: Things that work with Rootless Docker 20.10 and Rootless Podman 2.1 will work with Rootless Kubernetes as well. Other things will not.
+
+cgroup:
+- No support for cgroup v1.
+- [Hugepages](../20190129-hugepages.md) cannot be supported because systemd doesn't support delegation of the hugetlb controller: https://github.com/systemd/systemd/issues/16325
+- Device controller cannot be supported as well, but it is not a huge deal, because non-root users don't have permission to access insecure devices anyway.
+
+Network:
+- kube-proxy needs the following `KubeProxyConfiguration` to avoid hitting errors during setting `sysctl` values:
+```yaml
+conntrack:
+# Skip setting sysctl value "net.netfilter.nf_conntrack_max"
+  maxPerCore: 0
+# Skip setting "net.netfilter.nf_conntrack_tcp_timeout_established"
+  tcpEstablishedTimeout: 0s
+# Skip setting "net.netfilter.nf_conntrack_tcp_timeout_close"
+  tcpCloseWaitTimeout: 0s
+```
+- Some CNI plugins might not work. Flannel (VXLAN) is known to work.
+- Limited network performance due to the slirp4netns overhead.
+  **Mitigation:** Install [`lxc-user-nic` (SETUID binary)](https://github.com/rootless-containers/rootlesskit/blob/v0.14.2/docs/network.md#--netlxc-user-nic-experimental).
+- NodePort less than 1024 cannot be exposed. This is not a problem with the default `--service-node-port-range` configuration (30000-32767).
+  **Mitigation:** set `CAP_NET_BIND_SERVICE` file capability on `rootlesskit` binary.
+
+Volumes:
+- Block device volumes and (kernel-mode) NFS does not work, because user namespace only supports `tmpfs`, `bind`, and FUSE filesystems.
+  `emptyDir`, `hostPath`, `local`, and API volumes (`configMap`, `secret`, `downwardAPI`, ...) are known to work without any issue.
+  FUSE-based CSI volumes can be supported as well.
+  **Mitigation:** Implement CSI plugins for FUSE implementation of NFS. Or implement a "proxy" CSI plugin that talks to a privileged CSI plugin daemon.
```

Comment:
how does local volumes work without rootless kubelet? on mount kubelet performs recursive chown/chmod of local volumes and I suspect that won't be allowed.

Reply:
chown/chmod can work, as long as enough numbers of subuid and subgid are configured for the current user.